camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelEndpointAnnotator.java

```diff
@@ -91,10 +91,13 @@ private void extractSetValue(EndpointValidationResult result, Set<String> valida
             for (String entry : validationSet) {
                 String propertyValue = entry;
 
-                int propertyIdx = fromElement.indexOf(propertyValue);
+                int startIdxQueryParameters = fromElement.indexOf("?" + propertyValue);
+                startIdxQueryParameters = (startIdxQueryParameters == -1) ? fromElement.indexOf("&" + propertyValue) : fromElement.indexOf("?");
```

Comment:
this will never be called unless it has ? in the query. It's seems the catalog validation return success = true if the URI has no query parameters